streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -268,9 +274,13 @@ public void onPartitionsAssigned(final Collection<TopicPartition> assignment) {
             final long start = time.milliseconds();
             try {
                 if (streamThread.setState(State.PARTITIONS_ASSIGNED) == null) {
-                    return;
-                }
-                if (streamThread.assignmentErrorCode.get() == StreamsPartitionAssignor.Error.NONE.code()) {
+                    log.error("State transition from {} to PARTITIONS_ASSIGNED failed. " +
```

Comment:
Why is this an ERROR? If we receive a SHUTDOWN signal, it's just an "eager exit" to not create tasks IMHO. We might want to log a DEBUG though.

Would also update the message:
```
log.debug("Skipping task creation in rebalance because we are already in shutdown phase.");
```

Reply:
Sounds good to me.